### PR TITLE
WT-10469 Fix no timestamp tombstones not removing history store entries (#8727)

### PR DESCRIPTION
**Pull request by Sean04**
(cherry picked from commit https://github.com/wiredtiger/wiredtiger/commit/ede3bedaa1e5bfd03b51857608843c00dabc03d5)

### CODE REVIEW

File: src/reconcile/rec_col.c [hunk 1]
```diff
@@ -920,9 +911,19 @@ __wt_rec_col_fix(
...
+
                     if (!WT_TIME_WINDOW_IS_EMPTY(&upd_select.tw))
                         WT_ERR(__wt_rec_col_fix_addtw(session, r, entry, &upd_select.tw));
                 }
+
+                /*
+                 * When a tombstone without a timestamp is written to disk, remove any historical
+                 * versions that are greater in the history store for this key.
+                 */
+                if (upd_select.no_ts_tombstone && r->hs_clear_on_tombstone)
```
Comment (kommiharibabu) [inline src/reconcile/rec_col.c:923]: It should be `upd_select.ooo_tombstone`.

**Review by kommiharibabu — APPROVED** [review]
LGTM after fixing the compilation and test failure

**Review by quchenhao — APPROVED** [review]
lgtm

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**